agent/engine/task_manager.go

```diff
@@ -472,19 +499,29 @@ func (mtask *managedTask) releaseIPInIPAM() {
 // handles it.
 func (mtask *managedTask) handleStoppedToRunningContainerTransition(status api.ContainerStatus, container *api.Container) {
 	containerKnownStatus := container.GetKnownStatus()
-	if status <= containerKnownStatus && containerKnownStatus == api.ContainerStopped {
-		if status.IsRunning() {
-			// If the container becomes running after we've stopped it (possibly
-			// because we got an error running it and it ran anyways), the first time
-			// update it to 'known running' so that it will be driven back to stopped
-			mtask.unexpectedStart.Do(func() {
-				seelog.Warnf("Managed task [%s]: stopped container [%s] came back; re-stopping it once",
-					mtask.Arn, container.Name)
-				go mtask.engine.transitionContainer(mtask.Task, container, api.ContainerStopped)
-				// This will not proceed afterwards because status <= knownstatus below
-			})
-		}
+	if status > containerKnownStatus {
+		// Event status is greater than container's known status.
+		// This is not a backward transition, return
+		return
+	}
+	if containerKnownStatus != api.ContainerStopped {
+		// Container's known status is STOPPED. Nothing to do here.
```

Comment:
did you mean to put container's known status is NOT stopped here?

Reply:
yes. thanks for catching that.